README.rst

```diff
@@ -24,72 +28,86 @@ RNA-Seq Normalization
+- UQ_ (Upper quartile)
+- CUF_ (Counts adjusted with UQ factors)
+- TMM_ (Trimmed mean of M-values)
+- CTF_ (Counts adjusted with TMM factors)
 
-.. _Wikipedia: https://en.wikipedia.org/wiki/Quantile_normalization
 
-Usage
-=====
+.. _FPKM: https://www.nature.com/articles/nmeth.1226
+.. _TPM: https://link.springer.com/article/10.1007/s12064-012-0162-3
+.. _UQ: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-94
+.. _CUF: https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02568-9/
+.. _TMM: https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25
+.. _CTF: https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02568-9/
 
-Install ``rnanorm`` Python package:
+Features
+========
 
-.. code::
+- Pure Python implementation (no need for R, etc.)
+- Scikit-learn_ compatible
+- Command line interface
+- Verbose documentation_ (at least we hope so...)
+- Tested methods
 
-    pip install rnanorm
 
-See ``rnanorm`` command help:
+.. _Scikit-learn: https://scikit-learn.org/
+.. _documentation: https://rnanorm.readthedocs.io/
 
-.. code::
 
-    rnanorm --help
+Install
+=======
 
-Run ``rnanorm`` with pre-computed gene lengths:
+We recommend installing RNAnorm with pip::
 
-.. code::
+   pip install rnanorm
 
-    rnanorm expr.tsv --cpm-output=expr.cpm.tsv --tpm-output=expr.tpm.tsv --gene-lengths=lengths.tsv
 
-Run ``rnanorm`` with genome annotation - gene lengths will be computed on the fly:
+Quick start
+===========
 
-.. code::
+Implemented methods can be used from Python or from the command line.
 
-    rnanorm expr.tsv --cpm-output=expr.cpm.tsv --tpm-output=expr.tpm.tsv --annotation=annot.gtf
+Normalize from Python
+---------------------
 
-For quantile normalization we suggest using TPM expressions on the input:
+Most commonly normalization methods are run from Python. E.g.::
 
-.. code::
+   >>> from rnanorm.datasets import load_rnaseq_toy
+   >>> from rnanorm import FPKM
+   >>> dataset = load_rnaseq_toy()
+   >>> dataset.exp
+          G1     G2      G3      G4       G5
+   S1  200.0  300.0   500.0  2000.0   7000.0
+   S2  400.0  600.0  1000.0  4000.0  14000.0
+   S3  200.0  300.0   500.0  2000.0  17000.0
+   S4  200.0  300.0   500.0  2000.0   2000.0
+   >>> fpkm = FPKM(dataset.gtf_path).set_output(transform="pandas")
+   >>> fpkm.fit_transform(dataset.exp)
+             G1        G2        G3        G4        G5
+   S1  100000.0  100000.0  100000.0  200000.0  700000.0
+   S2  100000.0  100000.0  100000.0  200000.0  700000.0
+   S3   50000.0   50000.0   50000.0  100000.0  850000.0
+   S4  200000.0  200000.0  200000.0  400000.0  400000.0
 
-    rnanorm expr.tpm.tsv --quantile-output=expr.quantile.tsv
 
-Contributing
-============
+Normalize from command line
+---------------------------
 
-Install ``rnanorm`` Python package for development:
+Often it is handy to do normalization from the command line::
 
-.. code::
+   rnanorm fpkm exp.csv --gtf annotation.gtf --out exp_fpkm.csv
```

Comment:
Explain how the input file should look like, i.e., the separator, does it need index, column names, etc. Consider showing how to transform the gtex expression matrix supplied with the package. Show multiple examples so that it is clear, which transformations need the transcriptomic reference and which do not. Show how to use the `--help` flag. It provides information based on the context, e.g., `rnanorm tpm --help` will let you know that you need to supply a GTF file as well.